### ISSUE TITLE
Add Link Option `data-sveltekit-invalidate`

### ISSUE THREAD

**Issue by lemmon** [sveltejs/kit#10274]
### Describe the problem

My client noted that when clicking the same link (user is already on) they expect the page data to reload. I must agree that in certain cases it makes sense. SvelteKit currently does nothing when user is clicking the same link over and over.

### Describe the proposed solution

I propose to add `data-sveltekit-invalidate` link option. When user is navigating on some other page nothign will change from current system. Howerver when user clicks on the same link again and the `data-sveltekit-invalidate` option is present SvelteKit triggers the page `load()` function again. Same thing when happens when navigating to other page.

### Alternatives considered

I have tried adding `on:click` event to links:

```js
function handleClick() {
  if (this.href === $page.url.href) {
    invalidate('page:data')
  }
}
```

This almost works, however the load function runs twice. I asume that one runs for the invalidate event and the other runs for the navigation event.

Another try:

```js
function handleClick() {
  if (this.href === $page.url.href) {
    goto(this.href, {
      replaceState: true,
      invalidateAll: true,
    })
  }
}
```

This almost works too however invalidating all the resources is not the desired outcome.

### Importance

would make my life easier

**Comment by dummdidumm** [comment]
I don't think a link option will suffice here - what exactly should be invalidated? If it's not everything but specific things - like in your case - it makes more sense to solve this programmatically.
Possibly we need an `invalidate` option for `goto` which would allow you to refresh things more granularly.
Another possibility is that you add a click handler to the `a` tag and do `preventDefault/stopPropagation` on it if the url is the same and do `invalidate` instead; shouldn't rerun twice then.

**Comment by lemmon** [comment]
I agree that this could be done with combination of goto and invalidate functionality. I don't like the syntax for this solution too much. Applying custom click event and overwriting kit's core functionality is't very pretty, in my opinion. Perhaps a tag `data-sveltekit-invalidate` would re-trigger only page load function but `data-sveltekit-invalidate-all` or `data-sveltekit-invalidate="all"` would trigger all the load functions. Or perhaps `data-sveltekit-invalidate="/something:something"` would run equivalent of running `invalidate('/something:something')`. With one exception, this event would seem as if user is navigating and the `$navigating` store would be triggered as well.

It's something to think about. I am personally fine with only limited functionality. No need to overthink it. Limited functionality might be better than none at all.

**Comment by steve-taylor** [comment]
Here's my workaround. In the root layout:

```ts
let sequence = 0

onNavigate(navigation => {
    if (navigation.to?.url.toString() === navigation.from?.url.toString()) {
        ++sequence
    }
})
```

And then I wrap the part of the layout I want to reload in `{#key sequence}{/key}`. This could simply be your `<slot />` or if, like me, you want to reset your layout's nav, you can broaden the `key` or even use multiple `key`s.

**Comment by Nic-Polumeyv** [comment]
Since #16572, clicking a link to the current URL re-runs all `load` functions and remote queries by default in 3.0. Per-link granular invalidation was not adopted, so if that is still wanted it should be a fresh issue.
